### PR TITLE
Hijack objectives will only be given out if there are 30 or more players.

### PR DESCRIPTION
**Pull request by DaxDupont**
[Changelogs]: 
:cl: Dax Dupont
tweak: Hijack objectives will only be given out if there are 30 or more players.
/:cl:

[why]: From Beesting: It's pretty dumb to encourage murderboning when there's not enough people to fight off the traitor.

I did it this way since it seems to consider hijacking an objective that needs to be counted before all others, that's why it's sensible to do it there instead of when adding the objective itself.

Closes #35186

### CODE REVIEW

**Comment by DaxDupont** [comment]
Also bee asked people to solve it for them so that's why I did so!

**Comment by improvedname** [comment]
Im gonna murderbone either way, my point still stands.

**Comment by DaxDupont** [comment]
@improvedname code cannot fix people being dicks.
It can only discourage them, which this will do since people won't feel as pressured to murderbone on lowpop for greentext.

File: code/modules/antagonists/traitor/datum_traitor.dm [hunk 1]
```diff
@@ -85,7 +85,9 @@
 	return
 
 /datum/antagonist/traitor/human/forge_traitor_objectives()
-	var/is_hijacker = prob(10)
+	var/is_hijacker = FALSE
+	if (GLOB.player_list.len > 30) // Less murderboning on lowpop thanks
```
Comment (lzimann) [inline code/modules/antagonists/traitor/datum_traitor.dm]: this needs to grab the amount of players actually playing the round

Comment (DaxDupont) [reply code/modules/antagonists/traitor/datum_traitor.dm]: @lzimann which glob is that
Also I euh considered that but there's usually plenty of late joins just after antags get assigned and people who AFK get disconnected after 15 mins anyways.

**Comment by BeeSting12** [comment]
thanks! ill close mine now

File: code/modules/antagonists/traitor/datum_traitor.dm (hunk 1 above)
Comment (lzimann) [reply code/modules/antagonists/traitor/datum_traitor.dm]: `GLOB.joined_player_list`

Comment (DaxDupont) [reply code/modules/antagonists/traitor/datum_traitor.dm]: done

**Comment by ExcessiveUseOfCobblestone** [comment]
Is there a reason this isn't tied to the hijack objective? Doesn't seem very ObjOri

**Comment by lzimann** [comment]
This is fine, think of it as a pre-condition of existence, so before the datum is even created, you would need to check if it can be created. Hijack is sort of a special objective, in the generation regard, so a special check is not that big of a deal.

**Comment by DaxDupont** [comment]
Yeah hijack seemed to be a special case, I was originally gonna do it
different but it seemed better to do it like this.

**Comment by PKPenguin321** [comment]
Again, maybe like 25 players? You don't need 30 to fight a traitor

**Comment by DaxDupont** [comment]
I'm happy with 30.

File: code/modules/antagonists/traitor/datum_traitor.dm [hunk 2]
```diff
@@ -85,7 +85,9 @@
 	return
 
 /datum/antagonist/traitor/human/forge_traitor_objectives()
-	var/is_hijacker = prob(10)
+	var/is_hijacker = FALSE
+	if (GLOB.joined_player_list.len > 30) // Less murderboning on lowpop thanks
```
Comment (lzimann) [inline code/modules/antagonists/traitor/datum_traitor.dm]: also this should probably be `>=`

Comment (DaxDupont) [reply code/modules/antagonists/traitor/datum_traitor.dm]: PR title is more than 30, but I can make it 30 and more, give me a second.

**Comment by PKPenguin321** [comment]
Why?

**Comment by DaxDupont** [comment]
Why 25 and not 30?

**Comment by DaxDupont** [comment]
Would it be more enjoyable for people to be forced into murderboning more or less for greentext?
They can still murderbone regardless. Revs is locked behind 30 players as well.

This only removes an incentive to murderbone under 30. Nothing else.

**Comment by PKPenguin321** [comment]
You could make a case for several of those numbers, but I don't think 30 people is necessary to stop a single murderer in most cases. 25 is sufficient enough, and it wouldn't be fair to servers like Terry that reach only around that number at highpop to never see this objective be utilized.

**Comment by DaxDupont** [comment]
Does terry even hit 25?

**Comment by lzimann** [comment]
The biggest problem is that most of the time with 25 pop, there will rarely be any security/head to stop a traitor. And at that point, all the antag really needs is a way to break into the armory and it's already good enough to kill every single person in the round. I doubt that 30 will make that much of a difference but at least the challenge is a bit more, well, challenging. If 30 proves to be too much, it's easy enough to tweak this value.

**Comment by JohnMcLaime** [comment]
A single person of any jobtitle can make or obtain the tools which will stop anyone, doesnt matter what gear they have (admin fuckery doesnt count), even if they have over 9000 tc from cargo nullies, a mech or whatever else obtainable by non admins. All you need to know is that such target exists and how to make 100 potawater grenade and make an explosive lance.

**Comment by 2Dollarbill** [comment]
Murderboning is just inherent to the game, you either make rules against it. I don't think anyone will stop a person from murderboning, and it depends on your outlook on it if you think it's bad. I don't like it, but I'm not gonna discourage people from doing it, it's their antag round.

**Comment by ExcessiveUseOfCobblestone** [comment]
> Yeah hijack seemed to be a special case, I was originally gonna do it different but it seemed better to do it like this.

remind me in the future to thunk about this

**Comment by DaxDupont** [comment]
It's just removing some encouragement. It's not gonna totally prevent murderboning, only encourage it less on lowpop.

**Pull request merged**